### PR TITLE
feat:[RP-21] - button with variants

### PR DESCRIPTION
**Pull request by dipSyxx**
# Button component with variants

## Ticket Link
https://rozetka-home-project.atlassian.net/browse/XXXXX

### CODE REVIEW

File: src/pages/index.tsx [hunk 1]
```diff
@@ -46,6 +49,34 @@ export const Home = () => {
                   </SwiperSlide>
                 ))}
               </MainSlider>
+              <AllShareMain>
+                <ButtonCustom
```
Comment (MaksPapirovnyk) [inline src/pages/index.tsx:53]: 1. it's just a `Button` component you don't need a postfix
2. props shouldn't have the prefix 'Button' it's clear, we know that it's button props
3. pass text as `children`
4. base className should be inside the Button
5. create a variant prop and pick up the appropriate class name from props

File: src/elements/AllShareMain/AllShareMain.tsx [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import React, { ReactNode } from 'react'
+
+import styles from '../AllShareMain/AllShare.module.sass'
+
+type ChildreProp = {
+  children: ReactNode
+}
+
+export const AllShareMain = ({ children }: ChildreProp) => {
```
Comment (MaksPapirovnyk) [inline src/elements/AllShareMain/AllShareMain.tsx:9]: What is this? is it a Button?

File: src/elements/Burger/BurgerMenu.tsx [hunk 3]
```diff
@@ -70,7 +70,7 @@ export const BurgerMenu = ({
         className={`burger_holder animated_burger ${openBurgerMenu ? 'show_burger' : ''} `}
         onClick={handlerCloseBurgerMenu}
       >
-        <div className="burger_window" onClick={hendlerStopPropagationMenu}>
+        <div className="burger_window" onClick={handlerStopPropagationMenu}>
```
Comment (MaksPapirovnyk) [inline src/elements/Burger/BurgerMenu.tsx:73]: Why there is a prevented click? that doing nothing

**Comment by MaksPapirovnyk** [comment]
1. Please update the pull request description
2. too many changes for a single pull request
3. when you create a base component such as the `Button` try at first to implement the component itself and then implement its usage of it in a separate PR
4. somehow there is don't merged typography
5. all iterations should be step by step, either it's hard to read the code and understand changes that make PR review difficult
6. Don't import styles to the root app component, you make it global, and pollute CSS main chunk, import styles only to the component that owns this styles

File: src/elements/Burger/BurgerMenu.tsx (hunk 3 above)
Comment (dipSyxx) [reply src/elements/Burger/BurgerMenu.tsx:73]: This prevented click disables propagation

File: src/elements/AllShareMain/AllShareMain.tsx (hunk 2 above)
Comment (dipSyxx) [reply src/elements/AllShareMain/AllShareMain.tsx:9]: when I fix it, I will delete it, I conducted tests

**Comment by MaksPapirovnyk** [comment]
Closing this PR reason of the mess, we will reopen with new button

**Pull request closed without merging**